tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -374,6 +375,42 @@ static String getRequestId(DetachedSpan detachedSpan) {
         throw new SafeIllegalStateException("Unknown span type", SafeArg.of("detachedSpan", detachedSpan));
     }
 
+    @Nullable
+    static <T> T getTraceLocalValue(TraceLocal<T> traceLocal) {
+        Trace maybeCurrentTrace = currentTrace.get();
+        if (maybeCurrentTrace == null) {
+            return null;
+        }
+
+        return (T) maybeCurrentTrace.getTraceState().getTraceLocals().get(traceLocal);
+    }
+
+    @Nullable
+    static <T> T getTraceLocalValue(TraceLocal<T> traceLocal, Supplier<T> initialValue) {
+        Trace maybeCurrentTrace = currentTrace.get();
+        if (maybeCurrentTrace == null) {
+            return null;
+        }
+
+        return (T) maybeCurrentTrace
+                .getTraceState()
+                .getTraceLocals()
+                .computeIfAbsent(traceLocal, _key -> initialValue.get());
```

Comment:
Let's eagerly transform the `Supplier<T>` into `Function<?, T>` in the TraceLocal, that way we don't need to allocate lambdas for each `get` invocation.